monitoring/mock_uss/scdsc/routes_injection.py

```diff
@@ -111,105 +114,162 @@ def inject_flight(flight_id: str) -> Tuple[str, int]:
                 )
             )
 
-    # Check for operational intents in the DSS
-    start_time = scd.start_of(req_body.operational_intent.volumes)
-    end_time = scd.end_of(req_body.operational_intent.volumes)
-    area = scd.rect_bounds_of(req_body.operational_intent.volumes)
-    alt_lo, alt_hi = scd.meter_altitude_bounds_of(req_body.operational_intent.volumes)
-    vol4 = scd.make_vol4(
-        start_time, end_time, alt_lo, alt_hi, polygon=scd.make_polygon(latlngrect=area)
-    )
+    # Check if this is an existing flight being modified
+    while True:
+        with db as tx:
+            if flight_id in tx.flights:
+                # This is an existing flight being modified
+                existing_flight = tx.flights[flight_id]
+                if not existing_flight.locked:
+                    print(
+                        f"[inject_flight:{flight_id}] Existing flight locked for update"
+                    )
+                    existing_flight.locked = True
+                    break
+            else:
+                print(f"[inject_flight:{flight_id}] Request is for a new flight")
+                existing_flight = None
```

Comment:
I think with this implementation, there is a risk that two threads enter in the next section in parallel in the case of a new flight and creates conflicting/overriding actions. Creating a temporary locked record in the database may prevent the problem.

Reply:
Makes sense; I've added that placeholder and also ensured the other two endpoints respect this flight ownership concept (including the new placeholder).